UITests/PinnedTabsTests.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  PinnedTabsTests.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+
+class PinnedTabsTests: XCTestCase {
+    private var app: XCUIApplication!
+
+    override func setUpWithError() throws {
+        continueAfterFailure = false
+        app = XCUIApplication()
+        app.launchEnvironment["UITEST_MODE"] = "1"
+        app.launch()
+
+        app.typeKey("n", modifierFlags: .command)
+    }
+
+    override class func setUp() {
+        UITests.firstRun()
+    }
+
+    func testPinnedTabsFunctionality() {
+        openThreeSitesOnSameWindow()
+        openNewWindowAndLoadSite()
+        moveBackToPreviousWindows()
+
+        /// It is important that we wait some seconds to execute the navigate tabs commands
+        /// the reason is that switching windows take its time and the other windows is not show
+        /// immediately.
+        sleep(2)
```

Comment:
@jotaemepereira Have you tried to wait for a specific window to be visible? I realize it may be tricky, but we seem to have some code checking for specific webView title, e.g. this in BookmarksAndFavoritesTests:

```swift
app.windows.webViews[pageTitle]
```

Reply:
@ayoy I removed all the sleep methods and replaced them with correct `waitForExistance` calls.